fusequery/query/src/servers/server.rs

```diff
@@ -50,10 +50,9 @@ impl ShutdownHandle {
         let sessions = self.sessions.clone();
         let join_all = futures::future::join_all(shutdown_jobs);
         async move {
-            if !shutdown.load(Ordering::Relaxed) {
+            if Ok(false) == shutdown.compare_exchange(false, true, Ordering::SeqCst, Ordering::Acquire) {
```

Comment:
move this to `wait_for_termination_request` method and `impl Drop for ShutdownHandle`

by the way: maybe `if let Ok(false) = shutdown.compare_exchange(false, true, Ordering::SeqCst, Ordering::Acquire)` is better?

Reply:
yes, i will fix it, thanks